File: Javascript/AStar/spot.js
function Spot(i, j) {
	this.i = i;
	this.j = j;
	this.f = 0;
	this.g = 0;
	this.h = 0;
	this.neighbors = [];
	this.previous = undefined;
	this.wall = false;

	if (random(1) < 0.3) {
		this.wall = true;
	}

	this.show = function (clr) {
		fill(clr);
		noStroke();
		ellipse(this.i * w + w / 2, this.j * h + h / 2, w / 2, h / 2);
	};

	/*
  cols = 3
  rows = 3
    [
      [ (0,0), (0,1), (0,2) ],
      [ (1,0), (1,1), (1,2) ],
      [ (2,0), (2,1), (2,2) ]
    ]
  */

	this.addNeighbors = function (grid) {
		if (this.i < cols - 1) {
			this.neighbors.push(grid[this.i + 1][this.j]);
		}

		if (this.i > 0) {
			this.neighbors.push(grid[this.i - 1][this.j]);
		}

		if (this.j < rows - 1) {
			this.neighbors.push(grid[this.i][this.j + 1]);
		}

		if (this.j > 0) {
			this.neighbors.push(grid[this.i][this.j - 1]);
		}

		if (i > 0 && j > 0) {
			this.neighbors.push(grid[this.i - 1][this.j - 1]);
		}

		if (i < cols - 1 && j > 0) {
			this.neighbors.push(grid[this.i + 1][this.j - 1]);
		}

		if (i < cols - 1 && j < rows - 1) {
			this.neighbors.push(grid[this.i + 1][this.j + 1]);
		}

		if (i > 0 && j < rows - 1) {
			this.neighbors.push(grid[this.i - 1][this.j + 1]);
		}
	};
}

File: Python/AStar/spot.py
import pygame
import random
from settings import *

class Spot:
  def __init__(self, i,j):
    self.i = i
    self.j = j

    # A*
    self.f = 0
    self.g = 0
    self.h = 0

    self.neighbors = []
    self.previous = None
    self.wall = False

    if (random.random() < 0.4):
      self.wall = True


    self.display_serface = pygame.display.get_surface()

  def create_neighbors(self, grid):

      if (self.i < COLS - 1):
        self.neighbors.append(grid[self.i + 1][self.j])

      if (self.i > 0 ):
        self.neighbors.append(grid[self.i - 1][self.j])

      if (self.j < ROWS - 1):
        self.neighbors.append(grid[self.i][self.j + 1])

      if (self.j > 0 ):
        self.neighbors.append(grid[self.i][self.j - 1])

      # Corners

      if (self.i < COLS - 1 and self.j < ROWS - 1):
        self.neighbors.append(grid[self.i + 1][self.j + 1])

      if (self.i > 0 and self.j > 0):
        self.neighbors.append(grid[self.i - 1][self.j - 1])

      if (self.i < COLS - 1 and self.j > 0):
        self.neighbors.append(grid[self.i + 1][self.j - 1])

      if (self.i > 0  and self.j < ROWS - 1):
        self.neighbors.append(grid[self.i - 1][self.j + 1])

  def drawSpot(self,color):
    pygame.draw.ellipse(self.display_serface, color,
      pygame.Rect(
        self.i * SPOT_WIDTH + SPOT_WIDTH/2 + OFFSET,
        self.j * SPOT_HEIGHT + SPOT_HEIGHT/2 + OFFSET,
        SPOT_WIDTH / 2,
        SPOT_HEIGHT / 2
      )
    )
